### DISCUSSION TITLE
More organized GIT Commits

### DISCUSSION THREAD

**Discussion by ismaail** [Zakarialabib/myStockMaster#33] [Ideas]
Salam,

@Zakarialabib  in your projetcs, the way you push commits 
make it hard or impossible to follow the changes in the code and what the the commit is about.

This is bad if you want other develper to review your code and even contribute.

so what's the problem:
- the commit message like `update` does not tell what the commit is about 
- a single commit have 20 or 40 files changed!

Solutions:
- use temporary branch, after you finished all the small commits, then merge this branch into the master branch (see picture bellow)
- better commit messages.

![image](https://user-images.githubusercontent.com/4411670/216348267-96ad7abe-b23f-44a2-bace-77e0bbb9d4a9.png)

Thanks.

**Comment by Zakarialabib** [comment]
Wa 3alaykom salam, 
I wasn't organized lately, between different projects, i will work on improving commits
thank you for ur advices🙏👍🏼

**Comment by ismaail** [comment]
if you're using just command line for GIT, check out this GUI application [sublimemerge.](https://www.sublimemerge.com/)

**Comment by ronknight** [comment]
Make sure to pull any changes in the git repository first then sync changes to push it to the main branches.
